src/lib/core/adaptive/screen-type.ts

```diff
@@ -0,0 +1,134 @@
+  TABLET_PORTRAIT: '(min-width: 600px) and (max-width: 840px) and (orientation: portrait)',
+  WEB_PORTRAIT: '(min-width: 840px) and (orientation: portrait)',
+
+  HANDSET_LANDSCAPE: '(max-width: 960px) and (orientation: landscape)',
+  TABLET_LANDSCAPE: '(min-width: 960px) and (max-width: 1280px) and (orientation: landscape)',
+  WEB_LANDSCAPE: '(min-width: 1280px) and (orientation: landscape)',
+};
+
+export interface Breakpoint {
+  observe: Observable<boolean>;
+  destroy: () => void;
+}
+
+interface InternalBreakpoint {
+  query: string;
+  subject: BehaviorSubject<boolean>;
+  callback: (mql: MediaQueryList) => void;
+  usageCount: number;
+  watcher: MediaQueryList;
+}
+
+@Injectable()
+export class ScreenType {
+  private breakpoints: Map<string, InternalBreakpoint> = new Map();
+
+  constructor(private ngZone: NgZone) {}
+
+  /**
+   * Registers a breakpoint to be listened for.
+   */
+  registerBreakpoint(query: string): Breakpoint {
+    let breakpoint: InternalBreakpoint;
+    if (this.breakpoints.has(query)) {
+      breakpoint = this.breakpoints.get(query);
+      breakpoint.usageCount += 1;
+    } else {
+      prepareQueryCSS(query);
+      let watcher = window.matchMedia(query);
+      let callback = (mql: MediaQueryList) => {
+        this.ngZone.run(() => subject.next(mql.matches));
+      };
+      let subject = new BehaviorSubject(watcher.matches);
+      subject.subscribe(null, null, () => {
+        this.breakpoints.delete(query);
+      });
+      watcher.addListener(callback);
+
+      breakpoint = {
+        query: query,
+        subject: subject,
+        callback: callback,
+        watcher: watcher,
+        usageCount: 1
+      };
+      this.breakpoints.set(query, breakpoint);
+    }
+
+    return this.convertBreakpoint(breakpoint);
+  }
+
+  /**
+   * Converts an InternalBreakpoint to a Breakpoint, creating a destroy method to be called once.
+   */
+  private convertBreakpoint(breakpoint: InternalBreakpoint): Breakpoint {
+    let destroyed = false;
+    let destroy = (): void => {
+      if (destroyed) {
+        throw Error('Cannot make multiple calls to destroy breakpoint.');
+      }
+      if ((breakpoint.usageCount -= 1) <= 0) {
+        breakpoint.watcher.removeListener(breakpoint.callback);
+        breakpoint.subject.complete();
+        this.breakpoints.delete(breakpoint.query);
+      }
+      destroyed = true;
+    };
+
+    return {
+      destroy: destroy,
+      observe: breakpoint.subject.asObservable()
+    };
+  }
+}
+
+
+/**
+ * Private global registry for all dynamically-created, injected style tags
+ * @see prepare(query)
+ */
+const ALL_STYLES: {[key: string]: HTMLStyleElement} = {};
+
+
+/**
+ * For Webkit engines that only trigger the MediaQueryListListener
+ * when there is at least one CSS selector for the respective media query.
+ *
+ * @param query string The mediaQuery used to create a faux CSS selector
+ *
+ */
+function prepareQueryCSS(query: string) {
```

Comment:
I'd call this something like `insertEmptyStyleRuleForWebkitCompatibility`, and possible use the `Platform` service to gate. Then `ALL_STYLES` would be something like `styleElementForWebkitCompatibility` (not using CAPS_CASE because the value itself isn't a constant)